lib/src/Uncertainty/Distribution/TruncatedNormal.cxx

```diff
@@ -386,45 +438,76 @@ Point TruncatedNormal::computeCDFGradient(const Point & point) const
 
 /* Get the quantile of the distribution */
 Scalar TruncatedNormal::computeScalarQuantile(const Scalar prob,
-    const Bool tail) const
+                                              const Bool tail) const
 {
-  if (tail) return mu_ + sigma_ * DistFunc::qNormal(PhiBNorm_ - prob / normalizationFactor_);
-  return mu_ + sigma_ * DistFunc::qNormal(PhiANorm_ + prob / normalizationFactor_);
+  if (!(PhiANorm_ >= PhiBNorm_))
+    {
+      if (tail) return mu_ + sigma_ * DistFunc::qNormal(PhiBNorm_ - prob / normalizationFactor_);
+      return mu_ + sigma_ * DistFunc::qNormal(PhiANorm_ + prob / normalizationFactor_);
+    }
+  else
+    {
+      return DistributionImplementation::computeScalarQuantile(prob, tail);
+    }
 }
 
 /* Compute the mean of the distribution */
 void TruncatedNormal::computeMean() const
 {
-  mean_ = Point(1, mu_ - sigma_ * (phiBNorm_ - phiANorm_) * normalizationFactor_);
+  if (!(PhiANorm_ >= PhiBNorm_))
+    mean_ = Point(1, mu_ - sigma_ * (phiBNorm_ - phiANorm_) * normalizationFactor_);
+  else
+    {
+      const Scalar logphiA = DistFunc::dNormal(aNorm_);
+      const Scalar logphiB = DistFunc::dNormal(bNorm_);
```

Comment:
```suggestion
      const Scalar logphiA = DistFunc::logdNormal(aNorm_);
      const Scalar logphiB = DistFunc::logdNormal(bNorm_);
```

Reply:
what about this one @regislebrun ?

Reply:
Good catch. Fixed. Unfortunately it impacts all the moments...